45.md

```diff
@@ -29,29 +29,65 @@ In case a relay uses probabilistic counts, it MAY indicate it in the response wi
 
 Whenever the relay decides to refuse to fulfill the `COUNT` request, it MUST return a `CLOSED` message.
 
-## Examples
+## HyperLogLog
 
-### Followers count
+Relays may return an HyperLogLog value together with the count, hex-encoded.
 
 ```
-["COUNT", <subscription_id>, {"kinds": [3], "#p": [<pubkey>]}]
-["COUNT", <subscription_id>, {"count": 238}]
+["COUNT", <subscription_id>, {"count": <integer>, "hll": "<hex>"}]
 ```
 
+This is so it enables merging results from multiple relays and yielding a reasonable estimate of reaction counts, comment counts and follower counts, while saving many millions of bytes of bandwidth for everybody.
+
+### Algorithm
+
+This section describes the steps a relay should take in order to return HLL values to clients.
+
+1. Upon receiving a filter, if it has a single `#e`, `#p`, `#a` or `#q` item, read its 32th ascii character as a byte and take its modulo over 24 to obtain an `offset` -- in the unlikely case that the filter doesn't meet these conditions, set `offset` to the number 16;
+2. Initialize 256 registers to 0 for the HLL value;
+3. For all the events that are to be counted according to the filter, do this:
+    1. Read byte at position `offset` of the event `pubkey`, its value will be the register index `ri`;
+    2. Count the number of leading zero bits starting at position `offset+1` of the event `pubkey`;
```

Comment:
2. Count the number of leading zero bits starting at position 'offset+1' of the event 'pubkey' and add 1;

Reply:
Fixed.